controllers/components/kserve/kserve_controller_actions.go

```diff
@@ -271,3 +261,47 @@ func customizeKserveConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRe
 
 	return nil
 }
+
+func patchOwnerReference(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	k, ok := rr.Instance.(*componentsv1.Kserve)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Kserve)", rr.Instance)
+	}
+
+	l := logf.FromContext(ctx)
+	mc := &componentsv1.ModelController{}
+	if err := rr.Client.Get(ctx, client.ObjectKey{Name: componentsv1.ModelControllerInstanceName}, mc); err != nil {
+		if k8serr.IsNotFound(err) {
+			return odherrors.NewStopError("ModelController CR not exist: %v", err)
+		}
+		return odherrors.NewStopError("failed to get ModelController CR: %v", err)
+	}
+	for _, owners := range mc.GetOwnerReferences() {
+		if owners.UID == k.GetUID() {
+			return nil // kserve already as owner to modelcontroller, early exit
+		}
+	}
+
+	owners := []metav1.OwnerReference{}
+	owners = append(owners, mc.GetOwnerReferences()...) // keep the existing ones
```

Comment:
I think you need to check if the owner reference already exists, otherwise this may keep growing

Reply:
We may need to add a finalizer action to remove the owner reference

Reply:
yeah, but isnt that enojugh ? 
```
for _, owners := range mc.GetOwnerReferences() {
		if owners.UID == k.GetUID() {
			return nil // kserve already as owner to modelcontroller, early exit
		}
	}
```

Reply:
ah i see, so a different UID of the same one, and that could happen in a very short period.